astrodynamics-stk/src/main/java/org/b612foundation/adam/stk/propagators/StkSegmentPropagatedOrbit.java

```diff
@@ -196,8 +196,15 @@ void propagate(PropagationParameters propagationParams, JulianDate epoch, Julian
   }
 
   // TODO: make not Earth-specific
-  private static OrbitPointSummary buildFinalState(
+  private OrbitPointSummary buildFinalState(
```

Comment:
I think I'm following. It should return a final state that is: 

- an impact if there was an impact
- a close approach if it actually stopped propagation on a close approach (therefore there should always be at least one close approach)
- or the final propagated state

If that's correct then this is fine.

Reply:
That is correct. I've added javadoc to clarify.